rest/blip_sync.go

```diff
@@ -104,30 +103,35 @@ func (h *handler) handleBLIPSync() error {
 
 	// Create a BLIP context:
 	blipContext := blip.NewContext(BlipCBMobileReplication)
-	blipContext.Logger = DefaultBlipLogger(blipContext.ID)
 	blipContext.LogMessages = base.LogDebugEnabled(base.KeyWebSocket)
 	blipContext.LogFrames = base.LogDebugEnabled(base.KeyWebSocketFrame)
 
+	// Overwrite the logging context with the blip context ID
+	// Retaining other contextual information like db and username are not required for blip logging
+	h.db.Ctx = context.WithValue(h.db.Ctx, base.SGLogContextKey,
+		base.LogContext{CorrelationID: formatBlipContextID(blipContext.ID)},
```

Comment:
Per the docs, https://golang.org/pkg/context/#WithValue, the key shouldn't be a simple string to avoid potential collisions.  Based on example there, seems like we should at minimum define something like
type SGContextKey string, and then use SGContextKey(base.SGLogContextKey) here.

Reply:
Good find, thanks